orte/mca/rml/ofi/rml_ofi.h

```diff
@@ -0,0 +1,193 @@
+    struct fi_info *fabric_info;
+
+    /** Fabric Domain handle */
+    struct fid_fabric *fabric;
+
+    /** Access Domain handle */
+    struct fid_domain *domain;
+
+    /** Address vector handle */
+    struct fid_av *av;
+
+    /** Completion queue handle */
+    struct fid_cq *cq;
+
+    /** Endpoint to communicate on */
+    struct fid_ep *ep;
+
+    /** Endpoint name */
+    char ep_name[FI_NAME_MAX];
+
+    /** Endpoint name length */
+    size_t epnamelen;
+
+    /** OFI memory region */
+    struct fid_mr *mr_multi_recv;
+
+	/** buffer for tx and rx */
+    void *rxbuf;
+ 
+    uint64_t rxbuf_size;
+	
+	/* event,fd associated with the cq */ 
+	int fd;
+
+	/*event associated with progress fn */
+	opal_event_t progress_event;
+    bool progress_ev_active;
+
+    struct fi_context rx_ctx1;
+
+} ofi_transport_conduit_t;
+
+
+typedef struct {
+    struct orte_rml_base_module_t super;
+
+    /** Fabric info structure of all supported transports in system **/
+    struct fi_info *fi_info_list;
+
+   /** OFI ep and corr fi_info for all the transports (conduit) **/
+    ofi_transport_conduit_t ofi_conduits[MAX_CONDUIT];
+
+    size_t min_ofi_recv_buf_sz;
+
+    /** "Any source" address */
+    fi_addr_t any_addr;
+
+    /** number of Conduits currently opened **/
+    uint8_t conduit_open_num;
+
+    /** Unique message id for every message that is fragmented to be sent over OFI **/
+    uint32_t    cur_msgid;
+
+    opal_list_t     recv_msg_queue_list;     
+    opal_list_t     exceptions;
+    opal_list_t     queued_routing_messages;
+    opal_event_t    *timer_event;
+    struct timeval  timeout;
+} orte_rml_ofi_module_t;
+
+
+ORTE_MODULE_DECLSPEC extern orte_rml_component_t mca_rml_ofi_component;
+extern orte_rml_ofi_module_t orte_rml_ofi;
+
+int orte_rml_ofi_enable_comm(void);
+void orte_rml_ofi_fini(void);
+int orte_rml_ofi_ft_event(int state);
+int orte_rml_ofi_query_transports(opal_value_t **providers);
+int orte_rml_ofi_send_buffer_transport_nb(int conduit_id,
+                                              orte_process_name_t* peer,
+                                              struct opal_buffer_t* buffer,
+                                              orte_rml_tag_t tag,
+                                              orte_rml_buffer_callback_fn_t cbfunc,
+                                              void* cbdata);
+int orte_rml_ofi_send_transport_nb(int conduit_id,
+                                              orte_process_name_t* peer,
+                                              struct iovec* iov,
+                                              int count, 
+                                              orte_rml_tag_t tag,
+                                              orte_rml_callback_fn_t cbfunc,
+                                              void* cbdata);
+void free_conduit_resources( int conduit_id);
+void print_provider_list_info (struct fi_info *fi );
+
+
+//int orte_rml_ofi_progress(ofi_transport_conduit_t*);
```

Comment:
This doesn't need to be here -- the function is static inline, below (which is likely why it's commented out in this header file, and therefore should likely be removed).